.flake8

```diff
@@ -1,3 +1,46 @@
 [flake8]
 # Matching black's default
 max-line-length = 88
+
+extend-ignore =
+    D107,  # Missing docstring in __init__
+
+per-file-ignores = 
```

Comment:
@sigmavirus24 What do you think of this use of `per-file-ignores`? Versus, for example:

```
per-file-ignores = 
    twine/*:D1
    tests/*:D1,D205,D400
```

I was thinking this would make it easier for folks to fix issues, by deleting a specific code or file, then running `tox -e lint`.

Reply:
Why all of D1? We aren't violating _every_ error code in that class:

```
25    D100 Missing docstring in public module
12    D101 Missing docstring in public class
49    D102 Missing docstring in public method
83    D103 Missing docstring in public function
3     D104 Missing docstring in public package
8     D107 Missing docstring in __init__
```

Reply:
> We aren't violating _every_ error code in that class

True, but since only ones we aren't violating are D105 and D106, this felt like a reasonable compromise for read-/maintain-ability, compared to:

```
per-file-ignores =
    twine/__init__.py: D100,D101,D102,D103,D104
    twine/__main__.py: D100,D101,D102,D103,D104
    twine/_installed.py: D100,D101,D102,D103,D104
    # ...
```

I also thought it might encourage/aid adding all the missing docstrings for a single file at once.

Another option would be something like this:

```
per-file-ignores = 
    twine/*: D100,D101,D102,D103,D104
    twine/commands/*: D100,D101,D102,D103,D104
    tests/*: D100,D101,D102,D103,D104,D205,D400
```

I think that would encourage folks to fix a single error code across groups of files. I think that's also a reasonable approach, and I don't feel strongly.

Reply:
That second option looks good to me

Reply:
Done in https://github.com/pypa/twine/pull/634/commits/d2c1e5edf689dfdce92e35b5f89afc31a30adccf